modules/k8s/persistent_volume.go

```diff
@@ -0,0 +1,102 @@
+func ListPersistentVolumesE(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) ([]corev1.PersistentVolume, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clientset.CoreV1().PersistentVolumes().List(context.Background(), filters)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Items, nil
+}
+
+// GetPersistentVolume returns a Kubernetes PersistentVolume resource with the given name. This will fail the test if there is an error.
+func GetPersistentVolume(t testing.TestingT, options *KubectlOptions, name string) *corev1.PersistentVolume {
+	pv, err := GetPersistentVolumeE(t, options, name)
+	require.NoError(t, err)
+	return pv
+}
+
+// GetPersistentVolumeE returns a Kubernetes PersistentVolume resource with the given name.
+func GetPersistentVolumeE(t testing.TestingT, options *KubectlOptions, name string) (*corev1.PersistentVolume, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+	return clientset.CoreV1().PersistentVolumes().Get(context.Background(), name, metav1.GetOptions{})
+}
+
+// WaitUntilPersistentVolumeAvailableE waits until the given Persistent Volume is the 'Available' status,
+// retrying the check for the specified amount of times, sleeping
+// for the provided duration between each try.
+// This will fail the test if there is an error.
+func WaitUntilPersistentVolumeAvailable(t testing.TestingT, options *KubectlOptions, pvName string, retries int, sleepBetweenRetries time.Duration) {
+	require.NoError(t, WaitUntilPersistentVolumeAvailableE(t, options, pvName, retries, sleepBetweenRetries))
+}
+
+// WaitUntilPersistentVolumeAvailableE waits until the given PersistentVolume is in the 'Available' status,
+// retrying the check for the specified amount of times, sleeping
+// for the provided duration between each try.
+func WaitUntilPersistentVolumeAvailableE(
+	t testing.TestingT,
+	options *KubectlOptions,
+	pvName string,
+	retries int,
+	sleepBetweenRetries time.Duration,
+) error {
+	statusMsg := fmt.Sprintf("Wait for Persistent Volume %s to be available", pvName)
+	message, err := retry.DoWithRetryE(
+		t,
+		statusMsg,
+		retries,
+		sleepBetweenRetries,
+		func() (string, error) {
+			pv, err := GetPersistentVolumeE(t, options, pvName)
+			if err != nil {
+				return "", err
+			}
+			if !IsPersistentVolumeAvailable(pv) {
+				return "", NewPersistentVolumeNotAvailableError(pv)
+			}
+			return "Persistent Volume is now available", nil
+		},
+	)
+	if err != nil {
+		logger.Logf(t, "Timeout waiting for PersistentVolume to be available: %s", err)
+		return err
+	}
+	logger.Logf(t, message)
+	return nil
+}
+
+// IsPersistentVolume returns true if the given PersistentVolume is available
+func IsPersistentVolumeAvailable(pv *corev1.PersistentVolume) bool {
+	return pv != nil && pv.Status.Phase == corev1.VolumeAvailable
```

Comment:
I was wondering if should be checked for corev1.VolumeBound / corev1.VolumeReleased?
Can be use-cases when is required to check already bound PV

Reply:
My initial intention was to have a generic WaitForPersistentVolumeInPhase which would receive the Phase to wait for. Then I would have smaller wrapper functions for each state. 

However, at the end, and due to our requirements, I just implemented the 'Available' one, with the idea of implementing the others when I needed them.

I could implement the whole thing now though

Reply:
I think will be helpful to have a function that can get status as argument

Reply:
Done. 

I replaced the wait and test functions to check the 'available' status for generic ones which receive the status phase. I've named the status argument `pvStatusPhase` to make clear the relationship between the status and the type of argument the functions receive (`corev1.PersistentVolumePhase`, which is what's compared).

I think there's no need to add smaller wrapper functions for each phase since the generic functions are clear enough. Adding the functions would make the module unnecessarily long since there're 5 possible status phases (which means 10 extra functions).

Waiting for you comments 😊